examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -279,56 +249,43 @@ export class TaskList extends DataObject implements ITaskList {
 		}
 
 		// TODO: Delete any items that are in the root but missing from the external data
-		const updateTaskPs = updatedExternalData.map(async ([id, { name, priority }]) => {
-			// Write external data into savedData map.
-			const currentTask = this.savedData.get<PersistedTask>(id);
-			let incomingNameDiffersFromSavedName = false;
-			let incomingPriorityDiffersFromSavedPriority = false;
+		incomingExternalData.map(([id, { name: incomingName, priority: incomingPriority }]) => {
```

Comment:
```suggestion
		for (const [id, { name: incomingName, priority: incomingPriority }] of incomingExternalData) {
```
Double check me on the syntax. I'm not a fan of using `map` to iterate over something when we're not transforming it into something else. If we change this, the `return` below inside this block should become a `break`.